site/src/component/GradeDist/GradeDist.tsx

```diff
@@ -126,6 +126,14 @@ const GradeDist: FC<GradeDistProps> = (props) => {
 
     gradeDistData!
       .filter((entry) => {
+        if (
+          props.course &&
+          currentProf === 'ALL' &&
+          entry.department + ' ' + entry.courseNumber === props.course.department + ' ' + props.course.courseNumber
```

Comment:
why does this also add a course check? Is it supposed to be fixing some other bug?

If it's just the professor match, I would use logic consistent with the other spot, i.e. declare `const profMatch  = currentProf === 'ALL' || entry.instr...` and do one check `if (profMatch) { /* do stuff */ }`